selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/CheckSkipRootFolderTest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.selenium.factory;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.eclipse.che.api.workspace.shared.dto.ProjectConfigDto;
+import org.eclipse.che.selenium.core.SeleniumWebDriver;
+import org.eclipse.che.selenium.core.client.TestGitHubRepository;
+import org.eclipse.che.selenium.core.client.TestUserPreferencesServiceClient;
+import org.eclipse.che.selenium.core.factory.FactoryTemplate;
+import org.eclipse.che.selenium.core.factory.TestFactory;
+import org.eclipse.che.selenium.core.factory.TestFactoryInitializer;
+import org.eclipse.che.selenium.core.user.DefaultTestUser;
+import org.eclipse.che.selenium.core.webdriver.SeleniumWebDriverHelper;
+import org.eclipse.che.selenium.pageobject.Ide;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class CheckSkipRootFolderTest {
+
+  @Inject private TestFactoryInitializer testFactoryInitializer;
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private TestGitHubRepository testRepo;
+  @Inject private SeleniumWebDriver seleniumWebDriver;
+  @Inject private SeleniumWebDriverHelper seleniumWebDriverHelper;
+  @Inject private TestUserPreferencesServiceClient testUserPreferencesServiceClient;
+  @Inject private DefaultTestUser productUser;
+
+  @Inject
+  @Named("github.username")
+  private String gitHubUsername;
+
+  @Inject private Ide ide;
+
+  private TestFactory testFactory;
+  private String projectName;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    testUserPreferencesServiceClient.addGitCommitter(gitHubUsername, productUser.getEmail());
+    Path entryPath =
+        Paths.get(getClass().getResource("/projects/default-spring-project").getPath());
```

Comment:
Do we really need to add "default-spring-project" project into the test repository if we then adding into the factory template link to github project archive?

Reply:
Actually we need.
After adding the test repo - we have URLs like:
1) **SSH:** `git@github.com:gthub_user_name/name_repo`
2) **HTTPS:** `https://github.com/com:gthub_user_name/name_repo`
3) **ZIP:** `https://gthub_user_name/name_repo/archive/branch_name.zip`
So we use # 3 for the test

Reply:
In that case, IMHO, `testRepo.addContent(entryPath)` looks redundant.